users/views.py

```diff
@@ -29,65 +29,45 @@ def badge(request, person_id):
 
 
 def add_person(request):
-    try:
-        p_name = request.POST["name"]
-    except (KeyError, Name.DoesNotExist):
-        latest_person_list = Person.objects.order_by("-name")
-        context = {
-            "latest_person_list": latest_person_list,
-            "error_message":      "Invalid Person Name",
-        }
-        return render(request, "users/index.html", context)
-    else:
-        latest_person_list = Person.objects.order_by("-name")
-        for person in latest_person_list:
-            person_name = person.name
-            if p_name == person_name:
-                context = {
-                    "latest_person_list": latest_person_list,
-                    "error_message":      "Person already exists",
-                }
-                return render(request, "users/index.html", context)
-        p = Person(name=p_name)
-        p.save()
-        latest_person_list = Person.objects.order_by("-name")
-        context = {"latest_person_list": latest_person_list}
-        return render(request, "users/index.html", context)
+    # Construct context
+    page_context = {
+        'latest_person_list': Person.objects.order_by('-name'),
+        'error_message':      None,
+    }
+
+    person_name = request.POST.get('name', '')
+
+    if not person_name:
+        page_context['error_message'] = 'Username field must not be left blank'
+        return render(request, 'users/index.html', page_context)
+
+    new_person = Person(name=person_name)
+    new_person.save()
+
+    return redirect('users:index')
 
 
 def add_badge(request, person_id):
-    error = ''
+    # construct context
     person = get_object_or_404(Person, pk=person_id)
-    # Tries and excepts must be changed so no blank badges can be entered
-    try:
-        b_name = request.POST["name"]
-        b_presenter = request.POST["presenter"]
-    except (KeyError, Name.DoesNotExist):
-        return render(
-            request,
-            "users/badge.html",
-            {"person": person, "error_message": "Invalid Badge Name."},
-        )
-    except (KeyError, Presenter.DoesNotExist):
-        return render(
-            request,
-            "users/badge.html",
-            {"person": person, "error_message": "Invalid Presenter Name."},
-        )
+    page_context = {
+        'person': person,
+        'error_message': None,
+    }
+
+    # get values from the page form
+    badge_name = request.POST.get('name', '')
+    badge_presenter = request.POST.get('presenter', '')
+
+    # verify that form fields have be filled
+    if '' in (badge_name, badge_presenter):
```

Comment:
This is _logically_ fine, but I think it's more Pythonic to write this instead:
```suggestion
    if not (badge_name or badge_presenter):
```

Reply:
Agreed, I sometimes forget about the Truthiness of empty data structures